Source/Core/DolphinNoGUI/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+if(NOT(USE_X11 OR ENABLE_HEADLESS))
+  return()
+endif()
+
+if(APPLE)
+  set(DOLPHIN_EXE_BASE Dolphin)
+else()
+  set(DOLPHIN_EXE_BASE dolphin-emu)
+endif()
+
+set(DOLPHIN_NOGUI_EXE ${DOLPHIN_EXE_BASE}-nogui)
+
+set(NOGUI_SRCS MainNoGUI.cpp)
+
+add_executable(dolphin-nogui ${NOGUI_SRCS})
+set_target_properties(dolphin-nogui PROPERTIES OUTPUT_NAME ${DOLPHIN_NOGUI_EXE})
```

Comment:
I believe it doesn't need a different name on macOS. Just let it be dolphin-emu-nogui on all platforms so that it conforms to standard CLI tool naming.